File: classes.py
import random
from datetime import datetime

class PersonClass(object):
    random.seed(datetime.now())
    
# Constructor
    def __init__(self, number, beauty, intelligence, strength, PTI): 
        randNum = random.random()
        
        if(randNum < 0.01): #   1% Genetic Mutation 
            if(randNum < 0.002):
                self.beauty = (beauty + random.randint(0,20))%2
                self.intelligence = intelligence
                self.strength = strength
            elif(randNum < 0.006):
                self.intelligence = (intelligence + random.randint(0,80))%2
                self.beauty = beauty
                self.strength = strength
            else:
                self.strength = (strength + random.randint(0,100))%2
                self.beauty = beauty
                self.intelligence = intelligence
        else:
            self.beauty = beauty
            self.intelligence = intelligence
            self.strength = strength

        self.prone_to_illness = PTI
        self.health = 100
        self.fit = strength + beauty + intelligence
        self.surv = float(self.fit/20)
        self.id = number
        self.age = 0
        self.children = 0

File: julia/classes.jl
mutable struct People
    parent::Union{Nothing, People}

    beauty::Int64
    strength::Int64
    intelligence::Int64

    health::Int64
    fit::Int64
    surv::AbstractFloat
    mutation::AbstractFloat

    age::Int64
    children::Int64
    prone_to_illness::AbstractFloat

end


People(parent::People) = People(parent,0,0,0  ,0,0,0,   0,0,0,0)
People(num::Int64, num1::Int64) = People(nothing,0,0,0  ,0,0,num1, 0,  num,0,0)
People(beauty::Int64, strength::Int64, intelligence::Int64, r1,r2) = People(nothing, beauty,strength,intelligence, 100, beauty+strength+intelligence, (beauty+strength+intelligence)/20, r1, 0, 0, r2)